GUIDE.md

```diff
@@ -1,28 +1,160 @@
-# Server Endpoint
+# Guide
 
+ - The guiding focus to how to write a client to connect to a Tokyo server, receive information events and create actions to control your ship
+
+## 1. Connection information
+
+`ws://${host}/socket?key={key}&name{name}`
+
+| Parameters | Description |
+|--|--|
+| {host} | Domain or IP of server want to connect |
+| {key} | User's identity is unique and used to distinguish bots |
+| {name} | The display name will be shown on UI |
+
+When the WebSocket connection is established successfully, your bot is registered, displayed on web UI and ready to use.
+
+## 2. Events
+
+From WebSocket, the server consecutively sends events to the client every tick with the following structure.
+Event message structure contains all world info of the game, including players, bullets, dead person, etc.
+
+### 2.1. Event general structure
+
+```json
+{
+  "e": "state",
+  "data": {
+    "bounds": [1200.0, 800.0],
+    "players": [
+      {player},{player}
+    ],
+    "bullets": [
+      {bullet},{bullet}
+    ],
+    "dead": [
+      "respawn":{
+		    "secs_since_epoch":1631299999,
+	      "nanos_since_epoch":940876053
+       },
+       "player": {player}
+    ],
+    "scoreboard":{"0":100,"1":90,"2":80}
+  }
+}
 ```
-$HOST/socket
+
+| Fields | Description |
+|--|--|
+| e | Event information. The event from the server is always "state" |
```

Comment:
There're more than just state event. Namely, `id` and `teamnames`.

https://github.com/unrealhoang/tokyo-rs/blob/a2d4dd4cf6df4945809a1c2bab1fc6b7917846c2/client/src/models.rs#L36-L47